flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/catalog/DatabaseCalciteSchema.java

```diff
@@ -66,109 +68,154 @@
 class DatabaseCalciteSchema extends FlinkSchema {
 	private final String databaseName;
 	private final String catalogName;
-	private final Catalog catalog;
+	private final CatalogManager catalogManager;
 	// Flag that tells if the current planner should work in a batch or streaming mode.
 	private final boolean isStreamingMode;
 
-	public DatabaseCalciteSchema(String databaseName, String catalogName, Catalog catalog, boolean isStreamingMode) {
+	public DatabaseCalciteSchema(String databaseName, String catalogName, CatalogManager catalog, boolean isStreamingMode) {
 		this.databaseName = databaseName;
 		this.catalogName = catalogName;
-		this.catalog = catalog;
+		this.catalogManager = catalog;
 		this.isStreamingMode = isStreamingMode;
 	}
 
 	@Override
 	public Table getTable(String tableName) {
-		ObjectPath tablePath = new ObjectPath(databaseName, tableName);
+		ObjectIdentifier identifier = ObjectIdentifier.of(catalogName, databaseName, tableName);
+		return catalogManager.getTable(identifier)
+			.map(result -> {
+				CatalogBaseTable table = result.getTable();
+				if (result.isTemporary()) {
+					return convertTemporaryTable(new ObjectPath(databaseName, tableName), table);
+				} else {
+					return convertPermanentTable(
+						identifier.toObjectPath(),
+						table,
+						catalogManager.getCatalog(catalogName)
+							.flatMap(Catalog::getTableFactory)
```

Comment:
minor:  this feels a bit over complicated. How about just call `catalogManager.getCatalog(catalogName).getTableFactory()`?

Reply:
Your suggestion does not work. `catalogManager.getCatalog(catalogName)` returns `Optional`.